workspace/workspace.go

```diff
@@ -102,3 +103,53 @@ func (ws Workspace) Locate(exercise string) ([]string, error) {
 	}
 	return paths, nil
 }
+
+func (ws Workspace) SolutionPath(exercise, solutionID string) (string, error) {
+	paths, err := ws.Locate(exercise)
+	if !IsNotExist(err) && err != nil {
+		return "", err
+	}
+
+	return ws.DetermineSolutionPath(paths, exercise, solutionID, PathContainsSolution)
+}
+
+// TODO: PathContainsSolution might need a different name.
+func PathContainsSolution(path, solutionID string) (bool, error) {
+	s, err := NewSolution(path)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return s.ID == solutionID, nil
+}
+
+// TODO: DetermineSolutionPath might need a different name.
+func (ws Workspace) DetermineSolutionPath(paths []string, exercise, solutionID string, existsFn func(string, string) (bool, error)) (string, error) {
+	// Do we already have a directory for this solution?
+	for _, path := range paths {
+		ok, err := existsFn(path, solutionID)
+		if err != nil {
+			return "", err
+		}
+		if ok {
+			return path, nil
+		}
+	}
+	m := map[string]bool{}
+	for _, path := range paths {
+		m[path] = true
+	}
```

Comment:
If we haven't found a solution, then I want to be able to do a lookup to see whether or not we have a particular path. This creates a lookup table of the paths that were passed in.